calico_containers/pycalico/util.py

```diff
@@ -150,3 +150,44 @@ def validate_icmp_type(icmp_type):
     except ValueError:
         valid = False
     return valid
+
+def validate_hostname_port(hostname_port):
+    """
+    Validate the hostname and port format.  (<HOSTNAME>:<PORT>)
+    An IPv4 address is a valid hostname.
+
+    :return: Boolean: True if valid, False if invalid
+    """
+    # Should contain a single ":" separating hostname and port
+    hostname_port = hostname_port.split(":")
+    if len(hostname_port) != 2:
```

Comment:
Might be nice to have some debug logging in this function so we can track where validation went sour.